dashboard/ltiv1p3.py

```diff
@@ -0,0 +1,203 @@
+    try:
+        tool_config = ToolConfDict(ltiv1_p3)
+    except Exception as e:
+        raise e
+    # There should be one key per platform and the name relay on platforms generic domain not institution specific
+    platform_domain = list(ltiv1_p3.keys())[0]
+    client_id = ltiv1_p3[platform_domain][0]['client_id']
+    private_key_file_path = ltiv1_p3[platform_domain][0]['private_key_file']
+    public_key_file_path = ltiv1_p3[platform_domain][0]['public_key_file']
+    public_key = public_key_file_path if public_key_file_path else '/secret/public.key'
+    private_key = private_key_file_path if private_key_file_path else '/secret/private.key'
+    try:
+        private_key_content = open(private_key, 'r').read()
+        public_key_content = open(public_key, 'r').read()
+    except OSError as e:
+        raise e
+    tool_config.set_public_key(platform_domain, public_key_content, client_id=client_id)
+    tool_config.set_private_key(platform_domain, private_key_content, client_id=client_id)
+    return tool_config
+
+
+def get_jwks(request):
+    try:
+        tool_conf = get_tool_conf()
+    except (OSError, Exception) as e:
+        error_message = {
+            'lti_error_message': f'Fetching JWKS failed due to {e}'
+        }
+        return JsonResponse(error_message)
+    return JsonResponse(tool_conf.get_jwks())
+
+
+def check_if_instructor(roles, username, course_id):
+    teaching_roles = [role for role in roles if 'membership#Instructor' in role or
+                      'membership/Instructor#TeachingAssistant' in role]
+    if len(teaching_roles) == 0:
+        logger.info(f"user {username} is not instructor in course {course_id}")
+        return False
+
+    ta = [teach_role for teach_role in teaching_roles if 'membership/Instructor#TeachingAssistant' in teach_role]
+    if len(ta) != 0:
+        logger.info(f'user {username} is not an instructor of course {course_id}')
+        return False
+    else:
+        logger.info(f'user {username} is an instructor in course {course_id}')
+        return True
+
+
+def role_check(roles):
+    course_roles = [role for role in roles if 'membership' in role]
+    logger.info(course_roles)
+    user_roles = []
+    if len(course_roles) == 0:
+        institute_roles = [role for role in roles if "institution" in role]
+        logger.info(f'user dont seems to have role in Course must be admin {institute_roles}')
+        return user_roles
+
+    for role in course_roles:
+        split_ = role.split('#')
+        user_roles.append(split_[1])
+    logger.info(f'User role in course {user_roles}')
+    # this step eliminates duplicates of multiple same roles in the course Like TA can be instructor in multiple sections
+    user_roles = list(dict.fromkeys(user_roles))
+    # A TA can be instructor in other sections of the course but not in Lecture course so default to single role
+    user_roles = ['TeachingAssistant'] if not 'TeachingAssistant' else user_roles
```

Comment:
This looks very wrong.

Negating a static, non-empty string like `'TeachingAssistant'` **_always_** evaluates to `False`.  So the part of the statement following `else` will **_always_** be used.  This is equivalent to:

```
    user_roles = user_roles
```